src/Throttle/Throttler/LeakyBucketThrottler.php

```diff
@@ -25,89 +25,15 @@
 
 namespace Sunspikes\Ratelimit\Throttle\Throttler;
 
-use Sunspikes\Ratelimit\Cache\Exception\ItemNotFoundException;
-use Sunspikes\Ratelimit\Cache\Adapter\CacheAdapterInterface;
-use Sunspikes\Ratelimit\Time\TimeAdapterInterface;
-
-final class LeakyBucketThrottler implements RetriableThrottlerInterface
+final class LeakyBucketThrottler extends AbstractWindowThrottler implements ThrottlerInterface
```

Comment:
`(Leaky)Bucket` is a distinctly different algorithm from the `Window` type algorithms. I'm not sure this is going to function correctly, but if your intention is to create an abstract class for all throttlers, `AbstractThrottler` would (imo) be a better name.